### PR TITLE
chore(markdown): port marked to remarkable

### LINKED ISSUE #3728: Replace marked library with other Markdown parser
... it has (and had) some issues, and we can probably find something a bit more active.

- https://nodesecurity.io/advisories/101
- https://nodesecurity.io/advisories/23
- https://nodesecurity.io/advisories/24
- https://nodesecurity.io/advisories/22

On Test Pilot, we're currently using [**remarkable**](http://npm.im/remarkable) ([see usage](https://github.com/mozilla/testpilot/blob/master/gulpfile.js#L64-L73)), but not sure if its a 1:1 drop in replacement.

Semi-related, but we could also add a shame-badge from https://david-dm.org/mozilla/fxa-content-server to keep track of dependency "freshness" and confidence.

### PR DESCRIPTION
**Pull request by TDA**
Fixes #3728
 @pdehaan
@vladikoff

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -93,6 +94,7 @@
     "grunt-jsonlint": "1.0.7",
     "grunt-newer": "1.2.0",
     "grunt-nsp": "2.2.0",
+    "grunt-remarkable": "1.0.0",
```
Comment (pdehaan) [inline package.json]: Seems like we should remove **marked** and **grunt-marked** dependencies as well.

File: package.json [hunk 2]
```diff
@@ -76,6 +76,7 @@
     "universal-analytics": "0.3.11"
   },
   "devDependencies": {
+    "async": "2.0.0-rc.5",
```
Comment (pdehaan) [inline package.json]: related?
I dont see where this is used above.

File: package.json (hunk 1 above)
Comment (pdehaan) [inline package.json]: Hhmm, **marked** and **grunt-marked** were both dependencies, not devDependencies. Wonder if these need to be moved up to dependencies as well (which probably means regenerating the shrinkwrap file).

File: grunttasks/remarkable.js [hunk 3]
```diff
@@ -2,50 +2,41 @@
...
-  grunt.config('marked', {
+  grunt.config('remarkable', {
     options: {
       breaks: true,
-      gfm: true,
-      renderer: renderer,
-      sanitize: false
+      html: true,
+      linkify: false,
+      xhtmlOut: true,
```
Comment (pdehaan) [inline grunttasks/remarkable.js]: trailing comma (if we care)

**Comment by vladikoff** [comment]
```
Running "l10n-generate-tos-pp" task
Warning: Task "remarkable:tos_pp" not found. Use --force to continue.

```

File: package.json (hunk 1 above)
Comment (vladikoff) [inline package.json]: Yeap gotta do that ^

File: package.json [hunk 4]
```diff
@@ -24,6 +24,7 @@
   "author": "Mozilla (https://mozilla.org/)",
   "license": "MPL-2.0",
   "dependencies": {
+    "async": "2.0.0-rc.5",
```
Comment (pdehaan) [inline package.json]: Do we use this though?

File: package.json [hunk 5]
```diff
@@ -48,8 +49,8 @@
     "grunt-contrib-htmlmin": "1.3.0",
     "grunt-contrib-uglify": "1.0.1",
     "grunt-githash": "0.1.3",
-    "grunt-marked": "0.1.3",
     "grunt-po2json": "git://github.com/shane-tomlinson/grunt-po2json.git#2f415c8",
+    "grunt-remarkable": "1.0.0",
```
Comment (pdehaan) [inline package.json]: Update npm-shrinkwrap.json?

File: package.json (hunk 2 above)
Comment (TDA) [inline package.json]: its required for remarkable to work.

File: grunttasks/remarkable.js (hunk 3 above)
Comment (TDA) [inline grunttasks/remarkable.js]: removed

File: package.json (hunk 5 above)
Comment (TDA) [inline package.json]: updated

**Comment by TDA** [comment]
@vladikoff Updated shrikwrap :) passing tests. Final r?

File: package.json (hunk 4 above)
Comment (vladikoff) [inline package.json]: @TDA why is this here?

Comment (TDA) [inline package.json]: remarkable depends on it, but does not install it by itself. so we need this here

Comment (vladikoff) [inline package.json]: > remarkable depends on it, but does not install it by itself.

wat

Comment (TDA) [inline package.json]: @vladikoff  its weird, throws an error if able isnt there

Comment (TDA) [inline package.json]:
```

Loading "remarkable.coffee" tasks...ERROR
>> Error: Cannot find module 'async'
Loading "remarkable.js" tasks...ERROR
>> Error: Cannot find module 'async'

```

Comment (pdehaan) [inline package.json]: Can't repro locally on basic remarkable example from their README.md. Wonder if its Grunt specific...

```js
var Remarkable = require('remarkable');
var md = new Remarkable();

console.log(md.render('# Remarkable rulezz!'));
// => <h1>Remarkable rulezz!</h1>
```

Comment (vladikoff) [inline package.json]: Please remove it and rewrap and we will go from there

Comment (TDA) [inline package.json]: My bad, i should have been more clear, it is required by `grunt-remarkable` not by remarkable itself.
https://github.com/rwky/grunt-remarkable/blob/master/tasks/remarkable.coffee#L4
@vladikoff

Comment (pdehaan) [inline package.json]: Actually, I can reproduce this locally if I create a super-basic **grunt-remarkable** example:

```
➜  remarkable-async grunt remarkable
Loading "remarkable.coffee" tasks...ERROR
>> Error: Cannot find module 'async'
Loading "remarkable.js" tasks...ERROR
>> Error: Cannot find module 'async'
Warning: Task "remarkable" not found. Use --force to continue.

Aborted due to warnings.
```

I can't reproduce this with a plain Grunt example, or something that uses **grunt-nsp** which makes me suspicious of **grunt-remarkable**. Digging deeper I'm super doubtful of this:

[**rwky/grunt-remarkable** tasks/remarkable.coffee:4](https://github.com/rwky/grunt-remarkable/blob/a833d6324ae1c83190ab5fcfd26f1e5038444d7d/tasks/remarkable.coffee#L4):

```coffee
async = require 'async'
```

Yet, I don't see **async** defined in the package.json's dependencies or devDependencies:
https://github.com/rwky/grunt-remarkable/blob/a833d6324ae1c83190ab5fcfd26f1e5038444d7d/package.json#L12-L17 (and I don't think there's a built-in Node.js **async** module):

```json
  "dependencies": {
    "remarkable": "1.6.x"
  },
  "devDependencies": {
    "coffee-script":"1.10.x"
  },
```

So adding async manually seems to be a hacky workaround, but fixes the issue. I'll file a bug w/ grunt-remarkable and see if we can get a fix pushed to npm.

Comment (pdehaan) [inline package.json]: @TDA will submit a PR to the upstream **rwky/grunt-remarkable** repo so we don't have to dirty our package.json with **async** (or fork the repo and fix one dependency).

Comment (TDA) [inline package.json]: Da-Done! Once that gets merged, I will remove and re-wrap this and we should be good to go :)

**Comment by TDA** [comment]
@vladikoff @pdehaan  Updated with latest grunt-remarkable.

**Comment by pdehaan** [comment]
LGTM. r? @vladikoff @shane-tomlinson
